### DISCUSSION TITLE
List of all public GitHub Repositories for study purposes

### DISCUSSION THREAD

**Discussion by janvdhorst** [community/community#76994] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Hi everyone,

I am currently writing my masters thesis which includes the analysis of certain tools used in GitHub projects. To do this, I need a list of all public repositories that exist. 

Using GitHub API would take me around three weeks due to rate limitation

Is there any way I can get this kind of list a bit quicker?

Thank you

**Comment by jdtdm09** [comment]
GitHub Archive provides a record of public events that occur on GitHub, and it's available as a public dataset on Google BigQuery. You can use SQL queries to extract information from this dataset, including details about public repositories.

Here's a simplified example of how you might query the GitHub Archive dataset to get a list of public repositories:

SELECT
  repository_name
FROM
  `githubarchive.year.month`
WHERE
  type = 'CreateEvent'
  AND public = true

**Reply by CodingAndSharing** [reply]
The latest update from that github in bigquery at google has been last modified in January 2023. Is this source archived and not updated anymore in here? Or how often is updated?

**Comment by janvdhorst** [comment]
Thank you @jdtdm09 

I used the following bigQuery to analyse the usage of Dependabot

```
WITH Repositories AS (
  SELECT
    DISTINCT repo.name
  FROM
    `githubarchive.year.2022`
  WHERE type= 'PushEvent'
  AND EXTRACT(YEAR FROM created_at) = 2022
),

DependabotPRs AS (
  SELECT
    DISTINCT repo.name
  FROM
    `githubarchive.year.2022`
  WHERE
    type = 'PullRequestEvent'
    AND actor.login = 'dependabot[bot]'
    AND JSON_EXTRACT_SCALAR(payload, '$.action') = 'opened'
    AND EXTRACT(YEAR FROM created_at) = 2022
),
```

Which gave me 2.1 Million Repos with Dependabot for 2020 and 790.803 for 2022, which is less than half. 
Did Dependabot really become this unpopular in one year or did they change something about the usage?
